src/utils/utils.ts

```diff
@@ -27,22 +27,36 @@ export function getOrCreateTerminal(): vscode.Terminal {
     return globalState.maestroTerminal;
 }
 
-export async function updateYamlSchemaAssociations(schemaPath: string) {
+/**
+ * Updates the YAML schema associations in VS Code settings to ensure only our schema is used
+ * for Maestro YAML files. This prevents conflicts with other YAML schemas.
+ * Note: This function only manages the main extension schema patterns and does not affect
+ * test-specific schema configurations.
+ */
+export async function updateYamlSchemaAssociations(schemaPath: string = './schema/schema.v0.json') {
     const yamlConfig = vscode.workspace.getConfiguration('yaml');
-
     const currentSchemas = yamlConfig.get<{ [key: string]: string[] }>('schemas') || {};
-
     const filePatterns = getFilePatterns();
 
+    // Create updated schemas by removing ALL entries that point to our schema file
     const updatedSchemas = Object.fromEntries(
-        Object.entries(currentSchemas).filter(
-            ([, patterns]) => !patterns.some(pattern => filePatterns.includes(pattern))
+        Object.entries(currentSchemas).filter(([schema]) => {
+            // Remove any schema entry that points to our schema file, regardless of the path
+            return !schema.endsWith('schema.v0.json');
```

Comment:
Is this a dangerous match? This plugin doesn't have a very uniquely named schema.

Reply:
I had done an earlier match that matches the directory reads as maestro-workbench then ends with `schema.v0.json`.

We still have a problem. The yaml schema registered here https://github.com/Mastersam07/maestro-workbench/blob/dev/.vscode/settings.json gets added to the user settings also.